### DISCUSSION TITLE
Mitsuba in C++ domain

### DISCUSSION THREAD

**Discussion by buddhasystem** [mitsuba-renderer/mitsuba3#1242] [Q&A]
Please accept my apologies in case I overlooked something. One possibility in the project I'm currently working on is that I'll need to remain entirely in the C++ domain, when interfacing Mitsuba with the main app (particle transport). Does Mitsuba offer such a possibility? Any pointers to documentation and/or examples will be greatly appreciated. Thank you!

**Comment by njroussel** [comment]
Hi @buddhasystem 

To some extent, yes. The Python API is more complete, in fact, some stuff only exists in Python. I think that as long as you're just interested in traditional _forward_ rendering, you should be fine. Unfortunately, if you're looking to do use its inverse rendering capabilities it will require a lot more work as those features are almost all exclusively available in Python.

We don't have guides or documentation specifically for C++, however the Python bindings should be close to the original C++ interfaces.
